src/gui/processing/qgsprocessingmeshdatasetwidget.cpp

```diff
@@ -278,9 +292,25 @@ QWidget *QgsProcessingMeshDatasetGroupsWidgetWrapper::createWidget() SIP_FACTORY
 
 void QgsProcessingMeshDatasetGroupsWidgetWrapper::setWidgetValue( const QVariant &value, QgsProcessingContext &context )
 {
-  Q_UNUSED( context );
-  if ( mWidget )
-    mWidget->setValue( value );
+  if ( !mWidget )
+    return;
+
+  QList<int> datasetGroupIndexes;
+  if ( value.type() == QVariant::List )
+  {
+    //here we can't use  QgsProcessingParameters::parameterAsInts() because this method return empry list when first value is 0...
```

Comment:
```suggestion
    //here we can't use  QgsProcessingParameters::parameterAsInts() because this method return empty list when first value is 0...
```